distribution/lib/Standard/Database/0.0.0-dev/src/Connection/Redshift.enso

```diff
@@ -0,0 +1,79 @@
+import Standard.Database.Connection.SSL_Mode
+from Standard.Database.Connection.SSL_Mode import all
+import Standard.Database.Connection.Client_Certificate
+
+import Standard.Database.Connection.PostgreSQL
+
+polyglot java import com.amazon.redshift.jdbc.Driver
+
+type Redshift
+    ## Connect to a AWS Redshift database.
+
+       Arguments:
+       - host: The hostname of the database server (defaults to localhost).
+       - port: The port of the database server (defaults to 5432).
+       - schema: The schema to connect to (if not provided or empty, the default schema will be used).
+       - credentials: The credentials to use for the connection (defaults to PGPass or No Authentication).
+       - use_ssl: Whether to use SSL (defaults to `Require`).
+       - client_cert: The client certificate to use or `Nothing` if not needed.
+    type Redshift (host:Text) (port:Integer=5439) (schema:Text='') (credentials:Credentials|AWS_Profile|Nothing=Nothing) (use_ssl:(Disable|Require|Verify_CA|Full_Verification)=Require) (client_cert:Client_Certificate|Nothing=Nothing)
+
+    ## Build the Connection resource.
+
+       Arguments:
+       - options: Overrides for the connection properties.
+    connect : Connection_Options
+    connect options =
+        if Driver.isRegistered.not then Driver.register
+
+        properties = options.merge self.jdbc_properties
+        Connection.create_jdbc_connection self.jdbc_url properties self.dialect
+
+    ## Provides the jdbc url for the connection.
+    jdbc_url : Text
+    jdbc_url =
+        prefix = case self.credentials of
+            AWS_Profile _ -> 'jdbc:redshift:iam://'
+            _ -> 'jdbc:redshift://'
+        prefix + self.host + ':' + self.port.to_text + (if self.schema == '' then '' else '/' + self.schema)
+
+    ## Provides the properties for the connection.
+    jdbc_properties : [Pair Text Text]
+    jdbc_properties =
+        credentials = case self.credentials of
+            Nothing -> PostgreSQL.read_pgpass self.host self.port self.schema
+            AWS_Profile profile -> if profile == '' then [] else [Pair 'profile' profile]
+            Credentials username password ->
+                [ Pair 'user' username, Pair 'password' password ]
+
+        ssl_properties = case self.use_ssl of
+            Disable -> [Pair 'ssl' 'false']
+            Require -> [Pair 'ssl' 'true', Pair 'sslfactory' 'com.amazon.redshift.ssl.NonValidatingFactory']
+            Verify_CA cert_file ->
+                if cert_file.is_nothing then [Pair 'ssl' 'true', Pair 'sslmode' 'verify-ca'] else
+                    [Pair 'ssl' 'true', Pair 'sslmode' 'verify-ca', Pair 'sslrootcert' (File.new cert_file).absolute.path]
+            Full_Verification cert_file ->
+                if cert_file.is_nothing then [Pair 'ssl' 'true', Pair 'sslmode' 'verify-full'] else
+                    [Pair 'ssl' 'true', Pair 'sslmode' 'verify-full', Pair 'sslrootcert' (File.new cert_file).absolute.path]
```

Comment:
This seems completely copied-over from Postgres. Maybe we can make it into a private helper function and re-use it here instead of copying the code?